File: rxoauth/src/main/java/cz/ackee/rxoauth/OAuthStore.java
package cz.ackee.rxoauth;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Store of oauth2 credentials
 */
public class OAuthStore {
    public static final String TAG = OAuthStore.class.getName();
    private static final String SP_NAME = "oauth2";
    private static final String ACCESS_TOKEN_KEY = "oath2_access_token";
    private static final String REFRESH_TOKEN_KEY = "oath2_refresh_token";
    private final SharedPreferences sp;


    public OAuthStore(Context ctx) {
        sp = ctx.getSharedPreferences(SP_NAME, Context.MODE_PRIVATE);
    }

    public OAuthStore(SharedPreferences sp) {
        this.sp = sp;
    }

    public void saveOauthCredentials(String accessToken, String refreshToken) {
        sp.edit().putString(ACCESS_TOKEN_KEY, accessToken)
                .putString(REFRESH_TOKEN_KEY, refreshToken)
                .apply();
    }

    public String getAccessToken() {
        return sp.getString(ACCESS_TOKEN_KEY, null);
    }

    public String getRefreshToken() {
        return sp.getString(REFRESH_TOKEN_KEY, null);
    }

    public void onLogout() {
        sp.edit().clear().commit();
    }

    public void saveOauthCredentials(ICredentialsModel credentials) {
        saveOauthCredentials(credentials.getAccessToken(), credentials.getRefreshToken());
    }

}

File: rxoauth/src/main/java/cz/ackee/rxoauth/OAuthStore.kt
package cz.ackee.rxoauth

import android.content.Context
import android.content.SharedPreferences

/**
 * Persistence store of OAuth credentials.
 */
class OAuthStore {

    companion object {
        private const val SP_NAME = "oauth2"
        private const val ACCESS_TOKEN_KEY = "oath2_access_token"
        private const val REFRESH_TOKEN_KEY = "oath2_refresh_token"
    }

    private val sp: SharedPreferences

    val accessToken: String?
        get() = sp.getString(ACCESS_TOKEN_KEY, null)

    val refreshToken: String?
        get() = sp.getString(REFRESH_TOKEN_KEY, null)

    constructor(ctx: Context) {
        sp = ctx.getSharedPreferences(SP_NAME, Context.MODE_PRIVATE)
    }

    constructor(sp: SharedPreferences) {
        this.sp = sp
    }

    fun saveOauthCredentials(accessToken: String, refreshToken: String) {
        sp.edit()
                .putString(ACCESS_TOKEN_KEY, accessToken)
                .putString(REFRESH_TOKEN_KEY, refreshToken)
                .apply()
    }

    fun onLogout() {
        sp.edit().clear().apply()
    }

    fun saveOauthCredentials(credentials: OauthCredentials) {
        saveOauthCredentials(credentials.accessToken, credentials.refreshToken)
    }
}
